src/components/toast-notification-manager/daikin-toast-notification-manager.ts

```diff
@@ -0,0 +1,233 @@
+  private readonly _items!: readonly DaikinToastNotification[];
+
+  private _knownItemSet: ReadonlySet<DaikinToastNotification> = new Set();
+
+  private _containerRef = createRef<HTMLElement>();
+
+  get _positionY(): "top" | "bottom" {
+    return this.position.startsWith("top") ? "top" : "bottom";
+  }
+
+  private _close(target: DaikinToastNotification) {
+    const items = this._items;
+    const targetIndex = items.findIndex((item) => item === target);
+    const afterItems = items.filter((_, index) =>
+      this._positionY === "top" ? index > targetIndex : index < targetIndex
+    );
+
+    for (const item of items) {
+      item.style.removeProperty("--transition-duration");
+    }
+
+    target.hidden = true;
+
+    target.style.setProperty("--move-offset-x", TOAST_MOVE_OFFSET_X);
+    target.style.setProperty("--opacity", "0");
+    target.style.setProperty("--pointer-events", "none");
+
+    for (const item of afterItems) {
+      const height = item.clientHeight;
+      item.style.setProperty(
+        "--move-offset-y",
+        `calc(0.5rem + ${
+          height *
+          (this._positionY === "top"
+            ? TOAST_MOVE_OFFSET_Y_TOP_SIGN
+            : TOAST_MOVE_OFFSET_Y_BOTTOM_SIGN)
+        }px)`
+      );
+    }
+
+    setTimeout(() => {
+      for (const item of this._items) {
+        item.style.setProperty("--transition-duration", "0");
+      }
+
+      this.dispatchEvent(new CustomEvent("close", { detail: { target } }));
+      for (const item of this._items) {
+        item.style.removeProperty("--move-offset-y");
+      }
+    }, TOAST_DURATION);
+  }
+
+  private _handleClose(event: Event) {
+    event.stopPropagation();
+
+    this._close(event.target as DaikinToastNotification);
+  }
+
+  private _handleNewItem(newItem: DaikinToastNotification): void {
+    const container = this._containerRef.value;
+
+    if (!container) {
+      return;
+    }
+
+    requestAnimationFrame(() => {
+      newItem.style.setProperty("--opacity", "0");
+      container.style.setProperty(CONTAINER_TRANSITION_DURATION, "0");
+      container.style.setProperty(
+        CONTAINER_MOVE_OFFSET_Y,
+        `calc(0.5rem + ${
+          newItem.clientHeight *
+          (this._positionY === "top"
+            ? TOAST_MOVE_OFFSET_Y_TOP_SIGN
+            : TOAST_MOVE_OFFSET_Y_BOTTOM_SIGN)
+        }px)`
+      );
+
+      requestAnimationFrame(() => {
+        container.style.removeProperty(CONTAINER_TRANSITION_DURATION);
+        container.style.removeProperty(CONTAINER_MOVE_OFFSET_Y);
+        newItem.style.removeProperty("--opacity");
+
+        this._removeToast(newItem);
+      });
+    });
+  }
+
+  private _handleSlotchange() {
+    const items = this._items;
+
+    const newItems = items.filter((item) => !this._knownItemSet.has(item));
+    this._knownItemSet = new Set(items);
+    for (const newItem of newItems) {
+      this._handleNewItem(newItem);
+    }
+  }
+
+  private _removeToast(target: DaikinToastNotification) {
```

Comment:
As this function does not perform removal immediately, its name should indicate so.

```suggestion
  private _scheduleRemoveToast(target: DaikinToastNotification) {
```